### PR TITLE
Yeni bert classifier'i ekleyip service ve main binarylerine ekler.

### LINKED ISSUE #30: NLI -> BERT
NLI'yı bu modelle değiştiriyoruz: https://huggingface.co/deprem-ml/bert-fine-tuned-intent

### PR DESCRIPTION
**Pull request by telecastinator**
Yeni bert classifieri classifier interfaceinde class olarak yaptim, api servis ve main.py ye bagladim, commandline flaglere bagladim (hangi classifer calissin diye) ve evale ekledim.

Eski eksik eval datasiyla kurtarma bert classifierda performans soyle:

Intent: KURTARMA
==================================
KURTARMA Precision: 0.62
KURTARMA Recall: 1.00
KURTARMA F1: 0.77

Label eksikliginden geliyordur FPler ama ayni kosullarda kalsilastiirlinca rule based classifier hala karsilastirilabilir derecede iyi.

Closes #30

### CODE REVIEW

File: intent-classification-v0/ml_modules/bert_classifier.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    def __query(self, text):
+
+        response = requests.post(
+            API_URL, headers=headers, json={"inputs": text})
+        return response.json()
+
+    def all_intents(self):
+        """Returns list of all possible intents this classifier can classify."""
+        return ["ALAKASIZ", "KURTARMA", "BARINMA", "SAGLIK", "LOJISTIK", "SU", "YAGMA", "YEMEK", "GIYSI", "ELEKTRONIK"]
```
Comment (zafercavdar) [inline intent-classification-v0/ml_modules/bert_classifier.py:42]: Tuvalet is missing here.

**Review by zafercavdar — CHANGES_REQUESTED** [review]
If  there is any strict policy about not uploading data files, eval.csv should be removed. We also started using a new evaluation set covering examples from all classes. Merve posted it to AI channel last night.

File: intent-classification-v0/ml_modules/bert_classifier.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# Add logging.
+logging.basicConfig(level=logging.INFO)
+
+API_URL = "https://api-inference.huggingface.co/models/deprem-ml/bert-fine-tuned-intent"
+API_TOKEN = os.getenv("HF_HUB_TOKEN")
+
+headers = {"Authorization": f"Bearer {API_TOKEN}"}
+
+# The score threshold to deem a label as positive.
+CLASSIFICCTION_THRESHOLD = 0.5
```
Comment (zafercavdar) [inline intent-classification-v0/ml_modules/bert_classifier.py]: typo here

File: intent-classification-v0/ml_modules/bert_classifier.py (hunk 1 above)
Comment (telecastinator) [reply intent-classification-v0/ml_modules/bert_classifier.py:42]: Model (deprem-ml/bert-fine-tuned-intent) doesn't have TUVALET, just double checked this is the full list of intents it returns rn.

**Comment by telecastinator** [comment]
> If there is any strict policy about not uploading data files, eval.csv should be removed. We also started using a new evaluation set covering examples from all classes. Merve posted it to AI channel last night.

You're right, but I'm not aware of such policy, and the presence of other files in this data folder indicates it seems OK.

I know this eval file is dated and missing intents, and it is only provisionally here to have a workihng eval.py, I think it can stay.

We should definitely use the newest eval sets (as they get updated). Not sure if we'll keep keeping there here or hook up to a huggingface API.

File: intent-classification-v0/ml_modules/bert_classifier.py (hunk 2 above)
Comment (telecastinator) [reply intent-classification-v0/ml_modules/bert_classifier.py]: Acknowledged.

**Comment by telecastinator** [comment]
Oops mistakenly closed the PR.

**Review by yunusemreemik — COMMENTED** [review]
Need to fast convesation with this two PR. @ediziks @zafercavdar 
https://github.com/acikkaynak/depremadres-intent-classification-v0/pull/29

File: intent-classification-v0/app_main.py [hunk 3]
```diff
@@ -1,73 +1,66 @@
...
+    text: str
+
+class Response(BaseModel):
+    intents: List[str]
+
+
+@app.post("/get_intents/")
+async def Get_Intent(item: Request) -> Response:
+    if not item.text:
+        raise HTTPException(status_code=500, detail="Malformed request, no text")
```
Comment (zafercavdar) [inline intent-classification-v0/app_main.py]: client side error, I would return 400

File: intent-classification-v0/ml_modules/bert_classifier.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import logging
+import os
+import requests
+
+load_dotenv(".env")
+
+# Add logging.
+logging.basicConfig(level=logging.INFO)
+
+API_URL = "https://api-inference.huggingface.co/models/deprem-ml/bert-fine-tuned-intent"
```
Comment (zafercavdar) [inline intent-classification-v0/ml_modules/bert_classifier.py]: This model is outdated. "multilabel_earthquake_tweet_intent_bert_base_turkish_cased" is the updated one.

File: intent-classification-v0/ml_modules/bert_classifier.py (hunk 2 above)
Comment (zafercavdar) [inline intent-classification-v0/ml_modules/bert_classifier.py]:
```suggestion
CLASSIFICATION_THRESHOLD = 0.5
```

File: intent-classification-v0/ml_modules/bert_classifier.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    """
+    BERT based classifier to uses huggingface inference API to classify tweets.
+
+    Example usage:
+    >>> classifier = BertClassifier()
+    >>> classifier.classify("Kırıkkale'de deprem oldu.")
+    ["KURTARMA"]
+    """
+
+    def __init__(self, classification_threshold=CLASSIFICCTION_THRESHOLD):
```
Comment (zafercavdar) [inline intent-classification-v0/ml_modules/bert_classifier.py]:
```suggestion
    def __init__(self, classification_threshold=CLASSIFICATION_THRESHOLD):
```

**Review by zafercavdar — CHANGES_REQUESTED** [review]

**Comment by zafercavdar** [comment]
A general comment: is it possible to reduce the number of main.py files?

File: intent-classification-v0/ml_modules/bert_classifier.py (hunk 5 above)
Comment (telecastinator) [reply intent-classification-v0/ml_modules/bert_classifier.py]: Done.

File: intent-classification-v0/ml_modules/bert_classifier.py (hunk 4 above)
Comment (telecastinator) [reply intent-classification-v0/ml_modules/bert_classifier.py]: Done.

**Comment by telecastinator** [comment]
> A general comment: is it possible to reduce the number of main.py files?

I'm also not sure they exist but here's how I made sense; app_main.py is the main for the service, main.py is the command line version for quickly checking that the classifiers work. E.g 'python3 main.py --text="yardim"'. A quick way of integration testing of the models without needing to issue properly formed rest queries.

File: intent-classification-v0/app_main.py (hunk 3 above)
Comment (telecastinator) [reply intent-classification-v0/app_main.py]: Done

**Comment by telecastinator** [comment]
> Need to fast convesation with this two PR. @ediziks @zafercavdar
> #29

This looks great, I think there are no conflicts as is. Once #29 and #31 is merged, we can do a new PR to add a "KoalaClassifier" wrapper to the ml_modules and add it as a classifier option in the main service binary.

**Comment by telecastinator** [comment]
Oops, looks like max 2 reviewers are allowed.

**Comment by yunusemreemik** [comment]
> @ediziks @zafercavdar @telecastinator When you're available. I'll be waiting on discord/indent-classification.

**Review by zafercavdar — APPROVED** [review]

**Pull request merged**